docs/vc-http-api.yml

```diff
@@ -113,6 +113,65 @@ paths:
           description: invalid input!
         "500":
           description: error!
+  /presentations/available:
+    post:
+      tags:
+        - Holder
+      summary: Notifies a holder of an available presentation.
+      operationId: notifyPresentationAvailable
+      description:
+        This API is the first part of a holder to holder credential exchange over http.
+        The server MAY reply with a query for the client to proceed with a presentation.
+      requestBody:
+        description: Details the client provides to the server to help it decide if the presentation should be made.
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/NotifyPresentationAvailableRequest"
+
+      responses:
+        "200":
+          description: Proceed with presentation
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/NotifyPresentationAvailableResponse"
+        "400":
+          description: Request for presentation is malformed
+        "501":
+          description: Not implemented
+        "500":
+          description: internal error
+  /presentations/submissions:
+    post:
+      tags:
+        - Holder
+      summary: Provide a presentation for a holder to store.
+      operationId: storePresentation
+      description: Present from a client holder to a server holder, for the purpose of verification AND storage.
+      requestBody:
+        description: Details the client provides to the server to help it decide if the presentation should be made.
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/StorePresentationRequest"
+      responses:
+        "202":
+          description: Presentation accepted
+        "400":
+          description: Presentation is malformed
+        "401":
+          description: Presentation did not contain a proof
+        "402":
+          description: Payment required
+        "403":
+          description: Presentation verification failed
+        "425":
+          description: Server is unwilling to risk processing a request that might be replayed
```

Comment:
@OR13 does this mean there is no replay protection? If yes, then "Presentation is not replay protected" might be better. If not, why use this error if the only other reason for rejection would be that the holder is not trusted? Or can the submission come from a different endpoint/DID than the FYI notification to the potential Holder?

Reply:
I just included this error code since it seemed like it might bee relevant to certain veracity requirements... and because I knew it would spark a debate :)

even if the holder is authenticated, the holder might be presenting credentials which are "too young / unstable"... think finality requirements... its a thought I could live without it.

Reply:
@OR13 can you clarify? Do you mean "may be a replay of a previous credential" or "a request which may be replayed by the caller to update a presentation with a new state?" -- and if you mean the latter, that doesn't seem like a 400 error, but like a "201 Accepted" or "Continue" response, like "I got your message but I'm not donna deal with it now..."

Reply:
https://tools.ietf.org/html/rfc8470#section-5.2

```
  A 425 (Too Early) status code indicates that the server is unwilling
   to risk processing a request that might be replayed.

   User agents that send a request in early data are expected to retry
   the request when receiving a 425 (Too Early) response status code.  A
   user agent SHOULD retry automatically, but any retries MUST NOT be
   sent in early data.

   In all cases, an intermediary can forward a 425 (Too Early) status
   code.  Intermediaries MUST forward a 425 (Too Early) status code if
   the request that it received and forwarded contained an Early-Data
   header field.  Otherwise, an intermediary that receives a request in
   early data MAY automatically retry that request in response to a 425
   (Too Early) status code, but it MUST wait for the TLS handshake to
   complete on the connection where it received the request.

   The server cannot assume that a client is able to retry a request
   unless the request is received in early data or the Early-Data header
   field is set to "1".  A server SHOULD NOT emit the 425 status code
   unless one of these conditions is met.

   The 425 (Too Early) status code is not cacheable by default.  Its
   payload is not the representation of any identified resource.
```